src/test/test-alloc-util.c

```diff
@@ -37,20 +39,55 @@ static void test_alloca(void) {
 
 static void test_memdup_multiply(void) {
         int org[] = {1, 2, 3};
-        int *dup;
+        _cleanup_free_ int *dup;
 
-        dup = (int*)memdup_multiply(org, sizeof(int), 3);
+        dup = (int*) memdup_multiply(org, sizeof(int), 3);
 
         assert_se(dup);
         assert_se(dup[0] == 1);
         assert_se(dup[1] == 2);
         assert_se(dup[2] == 3);
-        free(dup);
+}
+
+static void test_memdup_suffix0_multiply(void) {
+        uint8_t org[] = {1, 2, 3};
```

Comment:
For the test to be more useful, this shouldn't be uint8_t, but something with a bigger size.